server/server_test.go

```diff
@@ -274,6 +276,121 @@ func TestSubmitCertificateVerification(t *testing.T) {
 	require.NoError(t, eg.Wait())
 }
 
+func createTestKeyFile(t *testing.T, dir, name string, data []byte) string {
+	path := filepath.Join(dir, name)
+	if err := os.WriteFile(path, data, 0o644); err != nil {
+		t.Fatalf("failed to write key file: %v", err)
+	}
+	return path
+}
+
+// Test submitting a challenge followed by proof generation and getting the proof via GRPC.
+func TestLoadTrustedKeysAndSubmit(t *testing.T) {
+	const keysNum = 2
+
+	// create test files
+	dir := t.TempDir()
+	t.Cleanup(func() { os.RemoveAll(dir) })
+
+	var (
+		trustedKeyCert *shared.OpaqueCert
+		trustedKey     []byte
+	)
+
+	challenge := []byte("challenge")
+
+	// User credentials
+	userPubKey, userPrivKey, err := ed25519.GenerateKey(rand.Reader)
```

Comment:
NIT: passing `nil` causes `GenerateKey` to use a cryptographically secure source for key generation. One only needs to pass a non-`nil` parameter if they want to ensure that specific keys are generated.

```suggestion
	userPubKey, userPrivKey, err := ed25519.GenerateKey(nil)
```